lib/rubocop/cop/style/symbol_proc.rb

```diff
@@ -22,10 +23,18 @@ class SymbolProc < Base
 
         def_node_matcher :proc_node?, '(send (const {nil? cbase} :Proc) :new)'
         def_node_matcher :symbol_proc?, <<~PATTERN
-          (block
-            ${(send ...) (super ...) zsuper}
-            $(args (arg _var))
-            (send (lvar _var) $_))
+          {
+            (block
+              ${(send ...) (super ...) zsuper}
+              $(args (arg _var))
+              (send (lvar _var) $_)
+            )
+            (numblock
+              ${(send ...) (super ...) zsuper}
+              $...
+              (send (lvar _var) $_)
+            )
+          }
```

Comment:
@marcandre I was trying to simplify this but can't find a way to do so that works because the numblock count node is just a bare integer, and I couldn't find a way to capture it inside a `{...}`. I opened https://github.com/rubocop-hq/rubocop-ast/issues/156.

Reply:
Closest I got, but fails because `integer?` is not defined on nodes

```ruby
def_node_matcher :symbol_proc?, <<~PATTERN
  ({block numblock}
    ${(send ...) (super ...) zsuper}
    ${(args (arg _)) #integer?}
    (send (lvar _var) $_))
PATTERN
```

Reply:
Fixed now, thanks @marcandre